singularity-eos/eos/eos_variant.hpp

```diff
@@ -267,6 +267,28 @@ class Variant {
         },
         eos_);
   }
+  template <typename RealIndexer, typename ConstRealIndexer>
+  inline void TemperatureFromDensityInternalEnergy(ConstRealIndexer &&rhos,
+                                                   ConstRealIndexer &&sies,
+                                                   RealIndexer &&temperatures,
+                                                   Real *scratch, const int num) const {
+    NullIndexer lambdas{}; // Returns null pointer for every index
+    return TemperatureFromDensityInternalEnergy(rhos, sies, temperatures, scratch, num,
+                                                lambdas);
+  }
+
+  template <typename RealIndexer, typename ConstRealIndexer, typename LambdaIndexer>
+  inline void
+  TemperatureFromDensityInternalEnergy(ConstRealIndexer &&rhos, ConstRealIndexer &&sies,
+                                       RealIndexer &&temperatures, Real *scratch,
+                                       const int num, LambdaIndexer &&lambdas) const {
+    return mpark::visit(
+        [&rhos, &sies, &temperatures, &scratch, &num, &lambdas](const auto &eos) {
+          return eos.TemperatureFromDensityInternalEnergy(rhos, sies, temperatures,
+                                                          scratch, num, lambdas);
+        },
+        eos_);
+  }
```

Comment:
Should the non-scratch vector calls be removed? Or perhaps just default scratch to a nullptr?

Reply:
Doesn't the python interface benefit from the non-scratch vector calls?

Reply:
Ah good point. :+1: